### DISCUSSION TITLE
Support for product reviews

### DISCUSSION THREAD

**Discussion by austriker27** [statamic-rad-pack/shopify#54] [Feature Requests]
# PRODUCT REVIEWS

I'd love the ability to add product reviews. I was surprised to see as Im setting up my shopify store that they aren't included out of the box. I know there's a ton of third party apps that offer the ability but it looks like standard might be Shopify's native app they built.

That being said, I reckon supporting shopify apps is a whole rabbit hole but I'd love to see if we can get basic support for the ability to have product reviews. 

Shopify's free app: https://apps.shopify.com/product-reviews

I haven't dived deep on it yet but if I find anything useful, I'll drop it in. 

The alternative would be something like stamped.io - From my limited exposure to it, I would think it would be as easy an dropping an iframe into a product page although it is likely the downside of this approach is less UI customization. 

# EDIT UPDATE

I installed the app on shopify here's their installation instructions:

> Add reviews to your product pages

> This code snippet displays reviews on your product pages and adds SEO-friendly review scores to your Google search results.

> Copy the following code snippet to your clipboard:
```
<div id="shopify-product-reviews" data-id="{{product.id}}">{{ product.metafields.spr.reviews }}</div>
```

>  Paste the snippet in your sections/product-template.liquid file where you want your reviews to appear. Usually this is just below the product.description tag, the code may look something like this: 
```
<div class="product-single__description rte" itemprop="description">
  {{ product.description }}
</div>

<div id="shopify-product-reviews" data-id="{{product.id}}">{{ product.metafields.spr.reviews }}</div>

{% if section.settings.show_share_buttons %}
  {% include 'social-sharing', share_title: product.title, share_permalink: product.url, share_image: product %}
{% endif %}
```

(There's also instructions for adding the stars to the shopify collections but because those aren't included (yet 🤞🏻) on this addon I've omitted them).

**Comment by jackabox** [comment]
The plugin you mentioned from Shopify stores a metafield on each product which contains the HTML to dump out the default product review. All of the actual reviews are stored on a custom endpoint that doesn't have public access so there is no direct way to pull these in.

This all works from the Shopify side (if you built a theme in it) as it then injects JS which calls the API and loads the data. This isn't possible through the Admin API/Storefront API. 

If you wanted to use an addon that is integrated on the Shopify side, you'd need to find something that had an API you could reference. 

One of the biggest stores that I know that use Shopify Storefront API (front-end) is [Victoria Beckham Beauty](https://www.victoriabeckhambeauty.com/) and they seem to be using [yotpo](https://www.yotpo.com/platform/reviews/) for the reviews so you may be able to look there for more answers.

**Reply by austriker27** [reply]
Gotcha, that's really helpful to know as I start looking at integrating reviews! I'm familiar with Yotpo as I've done some comparison shopping for review platforms. I'll dig into them more. And its helpful to have that Victoria Beckham Beauty website as reference. Thanks Jack!
